src/Common/Bridge/BridgeFacadeMethodsRule.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Bridge;
+
+use ArchitectureSniffer\SprykerAbstractRule;
+use PHPMD\AbstractNode;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\ClassAware;
+use ReflectionMethod;
+use ReflectionType;
+
+class BridgeFacadeMethodsRule extends SprykerAbstractRule implements ClassAware
+{
+    /**
+     * @var string
+     */
+    protected const RULE = 'Based on naming convention the bridge facade method must follow CRUD signature.';
+
+    /**
+     * @return string
+     */
+    public function getDescription()
+    {
+        return static::RULE;
+    }
+
+    /**
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return void
+     */
+    public function apply(AbstractNode $node)
+    {
+        if (
+            preg_match('([A-Za-z0-9]+Bridge$)', $node->getName()) === 0 ||
+            preg_match('#.*\\\\Dependency\\\\Facade.*#', $node->getNamespaceName()) === 0 ||
+            !$node instanceof ClassNode
+        ) {
+            return;
+        }
+        $bridgeName = $node->getFullQualifiedName();
+
+        foreach ($node->getMethods() as $method) {
+            $interfaceMethodName = sprintf('%s::%s', $bridgeName, $method->getName());
+            $interfaceMethodReflection = new ReflectionMethod($interfaceMethodName);
+            $methodReturnType = $interfaceMethodReflection->getReturnType();
+
+            if (preg_match('/^(get|read|find)/', $method->getName())) {
```

Comment:
I dont like the multiple words!
Looking at the tons of arch sniffer violations in our current master, we should really focus on 1 naming, and 1 alone.

So each of the CRUD methods MUST have one clear name, and the others will not be part of it IMO.
@gechetspr What do you think?
Whats the reason to allow multiple ones?

Also get vs find is a totally different concept (nullable) and therefore by design cannot be both under this rule.
Since the latter can return null, the first not.

Reply:
https://spryker.atlassian.net/wiki/spaces/RFC/pages/2922807829/ACCEPTED+Common+entity+manipulation+facade+CUD+function+signatures
https://spryker.atlassian.net/wiki/spaces/RFC/pages/2594832674?atlOrigin=eyJpIjoiNGIwMTFkYTIxNDY5NDNjMWJkMzJhNDIxZWVhMDZlZjMiLCJwIjoiY29uZmx1ZW5jZS1jaGF0cy1pbnQifQ

Reply:
I do not see "find" in the proposed signature

Following the proposed solution, the above functions would look like the following:


interface AclFacadeInterface
{ 
  public function getAclGroupCollection(AclGroupCriteriaTransfer $aclGroupCriteriaTransfer): AclGroupCollectionTransfer
    
  public function getAclRuleCollection(AclRuleCriteriaTransfer $aclRuleCriteriaTransfer): AclRuleCollectionTransfer
  
  public function getAclRoleCollection(AclRoleCriteriaTransfer $aclRoleCriteriaTransfer): AclRoleCollectionTransfer
}

interface AvailabilityFacadeInterface
{
  public function getProductConcreteAvailabilityCollection(ProductConcreteAvailabilityCriteriaTransfer $productConcreteAvailabilityCriteriaTransfer): ProductConcreteAvailabilityCollectionTransfer
}

interface ProductOfferFacade
{
  public function getProductOfferCollection(ProductOfferCriteriaTransfer $productOfferCriteriaTransfer): ProductOfferCollectionTransfer
}

Reply:
find means that we can have null in return. get returns transfer collection (it can be empty or not). you can also use get*Collection for findOne like universal method.